### ISSUE TITLE
Return 500-class error when transaction submission times out to provide more feedback to integratos

### ISSUE THREAD

**Issue by bartekn** [stellar/go-horizon#292]
I think it should return [`202 Accepted`](https://httpstatuses.com/202) instead. Developer would have to resubmit a transaction to check it's status.

**Comment by jedmccaleb** [comment]
But I think scott's point was that the person developing against horizon is going to have to handle the time out anyway (in cases of network failure etc) so why bother to have this extra error state?

**Comment by bartekn** [comment]
Because it's confusing. Connection reset may be caused by problems with Horizon but also by problems within the network of organization that's using Horizon. `202 Accepted` is a clear indication that the transaction was accepted but another request must be sent to check it's status.

**Comment by nullstyle** [comment]
I don't understand why you need to know that horizon accepted the transaction.  The fact that it accepted the transaction is zero guarantee that it will show up in the ledger.  While I might be convinced to presenting a 500-class error on timeout at horizon occurs because of consensus lag, I am adamantly against returning something that is in the 200-class:   Look at most code that checks for "was the request a success" and you'll see something naive like:

```
if code >= 200 && code < 300 {
  doSomeThingOnSuccess()
}
```

All that code will start behaving incorrectly if we return a 202 for transactions that will potentially fail.

**Comment by nullstyle** [comment]
It's worth noting that the whole point of horizon's transaction posting subsystem is to simplify integration... If you prefer to manage the state yourself, just post directly to stellar core and deal with all the complexity/power.

**Comment by bartekn** [comment]
Yeah, maybe `202 Accepted` is not a right choice. But I think returning 500-class error can be a good idea. My point is that connection reset gives you no idea what happened: ledger close delays/internal network issues/external network issues/horizon crash and so on.

**Comment by nullstyle** [comment]
Yep, I think that is valid.  It will reduce the instances of feedback-less errors while keeping the integration requirements the same.  I'll look into what it takes to fix later today.  IIRC, the choice to let the golang http layer time the connection out was made because that was the default behavior and required no work.
